pdns/dnsdist-rings.cc

```diff
@@ -19,35 +19,39 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
-#include "dnsdist.hh"
-#include "lock.hh"
+
+#include "dnsdist-rings.hh"
 
 size_t Rings::numDistinctRequestors()
 {
   std::set<ComboAddress, ComboAddress::addressOnlyLessThan> s;
-  ReadLock rl(&queryLock);
-  for(const auto& q : queryRing)
-    s.insert(q.requestor);
+  for (auto& shard : d_shards) {
```

Comment:
Maybe add `const` to the `shard` reference in all of your loops against `d_shards` (not in `dnsdist-rings.hh`) ?

Reply:
I would except I can't because I'm locking the mutex. I could make it mutable but that doesn't seem right here.